src/analyses/CMakeLists.txt

```diff
@@ -1,4 +1,9 @@
 file(GLOB_RECURSE sources "*.cpp" "*.h")
+list(REMOVE_ITEM sources
+    ${CMAKE_CURRENT_SOURCE_DIR}/reaching_definitions.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/reaching_definitions_without_sharing.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/reaching_definitions_with_sharing.cpp
+)
```

Comment:
This seems a bit sketchy. Why are you removing sources? If those shouldn't be part of the build, why are they here in the first place?

Reply:
The implementation is now templated and is contained in the header files which have an include statement at the bottom including those cpp files.

Reply:
If those files only contain template implementations, can you rename them to `.tpp` or something like that? This does have some history, and it'd clearly mark these files as "do not compile these separately", whereas removing them in the CMakeLists.txt is probably going to keep confusing everyone who reads this in the future.

Alternatively, if these implementations aren't terribly complex you could just directly copy them to the bottom of the headers and remove these .cpp files entirely.